### DISCUSSION TITLE
Code Review request for integration with CommandDotNet

### DISCUSSION THREAD

**Discussion by drewburlingame** [spectreconsole/spectre.console#605] [Q&A] [answered]
Hi all, I'm one of the maintainers of [CommandDotNet](https://github.com/bilal-fazlani/commanddotnet).  I'm also a fan of your AnsiConsole.  

Given your great design, I was able to integrate it as an optional extension for our framework, including our test tooling (with a couple of caveats}.  I was wondering if you'd be willing to review the significant files in [this PR](https://github.com/bilal-fazlani/commanddotnet/pull/362) and let me know if you think there will be problems or better ways to implement this. 

The significant files are

- CommandDotNet.Spectre/AnsiConsoleForwardingConsole.cs
- CommandDotNet.Spectre/SpectreMiddleware.cs
- CommandDotNet.Spectre.Testing/AnsiTestConsole.cs

and the tests to see it in action

next, I'm planning to integrate your prompting and obsolete the one I wrote.  I didnt put much time into it and it shows.  

Cheers,
Drew

**Comment by patriksvensson** [comment]
@drewburlingame Absolutely. I probably have some time this week to take a look at it.

**Comment by drewburlingame** [comment]
I've implemented the guts of the argument prompter to prompt for missing arguments here: https://github.com/bilal-fazlani/commanddotnet/pull/363  

Do you already have a method or a standard approach to prompting for a list of arguments that aren't predefined?  Think multi-selection but without preset choices to add.

I'm also having problems testing Confirm for bool values.  I've noted the failure here: https://github.com/bilal-fazlani/commanddotnet/pull/363/files#diff-3bf03d20a741f0c2b30ce5bc78b724cece304c7d39d1335b3c23bfbad9472fedR284

**Reply by drewburlingame** [reply]
update...

figured out the problem with confirming bool values. It was a bug in my missing arg prompter.  

also realized I need to use console.ReadKeyAsyc to mimic System.Console behavior.

**Comment by patriksvensson** [comment]
@drewburlingame Migrated issue to a discussion. Will try to take a look at the PR sometime today.

**Comment by drewburlingame** [comment]
I've tested this out in oue example program and it feels pretty smooth.  All I'm missing is a way to collect a list of values that are not predefined.  I could repeat a TextPrompt until an empty response is received.  Do you know of a better approach?

**Reply by drewburlingame** [reply]
this is the pattern I'm going with atm.  the prompt will exit when the last line is left blank.

```
fruit (Text)
> apple
> orange
> kiwi
> 
```

**Comment by drewburlingame** [comment]
@patriksvensson I'm going to go ahead and merge these so I can start using them.  I know you're busy and this isn't likely the highest priority.  If you find you have time and interest, you can comment on the closed PRs and I'll incorporate the feedback in a new branch.

**Reply by patriksvensson** [reply]
Sorry for disappearing there. Have been a hectic week with a sick family and a lot of work.
I've taken a look now, and I agree with @phil-scott-78 that it looks good! 👍

**Reply by drewburlingame** [reply]
No worries @patriksvensson.  Thank you for the review.

**Comment by phil-scott-78** [comment, marked as answer]
Fwiw I gave this a once over last night. I can't imagine I would have done anything much different. Looks quite nice

**Reply by drewburlingame** [reply]
thank you @phil-scott-78.  I appreciate the extra pair of eyes on it.
